hs_core/views/utils.py

```diff
@@ -499,16 +510,23 @@ def rename_irods_file_or_folder_in_django(resource, src_name, tgt_name):
             for fobj in res_file_objs:
                 old_str = fobj.resource_file.name
                 new_str = old_str.replace(src_name, tgt_name)
+                logical_file = fobj.logical_file if fobj.has_logical_file else None
                 fobj.delete()
-                ResourceFile.objects.create(content_object=resource, resource_file=new_str)
+                res_file = ResourceFile.objects.create(content_object=resource,
```

Comment:
There is tremendous confusion in ResourceFile handling elsewhere in the code that will impact this code. 

1. The ResourceFile object in your branch doesn't really support folders. I needed to work them into the structure with an external field "file_folder". See #1672. 
2. Other Filefields have names that are simple filenames, and the path to the name is a separate thing. This is automagical upon file upload. I forced this behavior by making file_folder a separate parameter in ResourceFile. Any other upload behavior requires subclassing FileField. See #1672 hs_core/models.py:get_path.

Thus, for consistency, the folder should be a separate parameter. my code would include resource_file={basename of new_str}, file_folder={everything but basename}. It is likely that I can make this automagical later by overriding ResourceFile.__init__ and automatically pushing the changed names into the real __init__. 

There is a systemic lack of consistency in the whole code base in handling of ResourceFiles, of which this is just one symptom. 

At the least, this will result in ResourceFile objects with variant structure (folder in name vs. folder in file_folder). Not particularly optimal.